### PR TITLE
make cache purge environment aware

### PR DESCRIPTION
**Pull request by jafowler**
This PR aims to make the zone cache functions environment aware. As of right now, the plan is to allow users to include zone versioning environments when they call their cache purge functions.
```
php vendor/bin/phpunit --configuration phpunit.xml
PHPUnit 5.7.27 by Sebastian Bergmann and contributors.

...............................................................  63 / 158 ( 39%)
............................................................... 126 / 158 ( 79%)
................................                                158 / 158 (100%)

Time: 4.8 seconds, Memory: 8.00MB

OK (158 tests, 983 assertions)

```

### CODE REVIEW

**Review by mgirouard — APPROVED** [review]
lgtm

File: tests/Endpoints/ZoneCacheTest.php [hunk 1]
```diff
@@ -75,4 +76,106 @@ public function testCachePurge()
...
+    {
+        $envResp = $this->getPsr7JsonResponseForFixture('Endpoints/getEnvironments.json');
+        $cacheResp = $this->getPsr7JsonResponseForFixture('Endpoints/cachePurge.json');
+        $mock = $this->getMockBuilder(\Cloudflare\API\Adapter\Adapter::class)->getMock();
+
+        $mock->expects($this->once())
+            ->method('get')
+            ->willReturn($envResp)
+            ->with(
+                $this->equalTo('zones/c2547eb745079dac9320b638f5e225cf483cc5cfdda41/environments'),
```
Comment (jacobbednarz) [inline tests/Endpoints/ZoneCacheTest.php:90]: php < 7.2 doesn't like commas here.

```suggestion
                $this->equalTo('zones/c2547eb745079dac9320b638f5e225cf483cc5cfdda41/environments')
```

File: tests/Endpoints/ZoneCacheTest.php [hunk 2]
```diff
@@ -75,4 +76,106 @@ public function testCachePurge()
...
+    {
+        $envResp = $this->getPsr7JsonResponseForFixture('Endpoints/getEnvironments.json');
+        $cacheResp = $this->getPsr7JsonResponseForFixture('Endpoints/cachePurgeEverything.json');
+        $mock = $this->getMockBuilder(\Cloudflare\API\Adapter\Adapter::class)->getMock();
+
+        $mock->expects($this->once())
+            ->method('get')
+            ->willReturn($envResp)
+            ->with(
+                $this->equalTo('zones/c2547eb745079dac9320b638f5e225cf483cc5cfdda41/environments'),
```
Comment (jacobbednarz) [inline tests/Endpoints/ZoneCacheTest.php:150]:
```suggestion
                $this->equalTo('zones/c2547eb745079dac9320b638f5e225cf483cc5cfdda41/environments')
```

File: src/Endpoints/Zones.php [hunk 3]
```diff
@@ -230,7 +240,10 @@ public function cachePurgeEverything(string $zoneID): bool
         return false;
     }
 
-    public function cachePurge(string $zoneID, array $files = null, array $tags = null, array $hosts = null): bool
+    /*
+     *  @SuppressWarnings(PHPMD)
+     **/
+    public function cachePurge(string $zoneID, array $files = null, array $tags = null, array $hosts = null, bool $includeEnvironments = false): bool
```
Comment (jacobbednarz) [inline src/Endpoints/Zones.php:246]:
```suggestion
    /**
     * @SuppressWarnings(PHPMD)
     */
    public function cachePurge(string $zoneID, array $files = null, array $tags = null, array $hosts = null, bool $includeEnvironments = false): bool
```

**Pull request merged**